### PR TITLE
jsonpb: handle empty strings for well-known types

### PR DESCRIPTION
**Pull request by cvanderschuere**
This changes jsonpb to treat empty strings as zero-values for Time and Duration well-known types

### CODE REVIEW

File: jsonpb/jsonpb_test.go [hunk 1]
```diff
@@ -761,12 +761,14 @@ var unmarshalingTests = []struct {
...
 	{"Duration with unicode", Unmarshaler{}, `{"dur": "3\u0073"}`, &pb.KnownTypes{Dur: &durpb.Duration{Seconds: 3}}},
 	{"null Duration", Unmarshaler{}, `{"dur":null}`, &pb.KnownTypes{Dur: nil}},
+	{"empty Duration", Unmarshaler{}, `{"dur":""}`, &pb.KnownTypes{Dur: &durpb.Duration{}}},
 	{"Timestamp", Unmarshaler{}, `{"ts":"2014-05-13T16:53:20.021Z"}`, &pb.KnownTypes{Ts: &tspb.Timestamp{Seconds: 14e8, Nanos: 21e6}}},
 	{"Timestamp", Unmarshaler{}, `{"ts":"2014-05-13T16:53:20Z"}`, &pb.KnownTypes{Ts: &tspb.Timestamp{Seconds: 14e8, Nanos: 0}}},
 	{"Timestamp with unicode", Unmarshaler{}, `{"ts": "2014-05-13T16:53:20\u005a"}`, &pb.KnownTypes{Ts: &tspb.Timestamp{Seconds: 14e8, Nanos: 0}}},
 	{"PreEpochTimestamp", Unmarshaler{}, `{"ts":"1969-12-31T23:59:58.999999995Z"}`, &pb.KnownTypes{Ts: &tspb.Timestamp{Seconds: -2, Nanos: 999999995}}},
 	{"ZeroTimeTimestamp", Unmarshaler{}, `{"ts":"0001-01-01T00:00:00Z"}`, &pb.KnownTypes{Ts: &tspb.Timestamp{Seconds: -62135596800, Nanos: 0}}},
 	{"null Timestamp", Unmarshaler{}, `{"ts":null}`, &pb.KnownTypes{Ts: nil}},
+	{"empty Timestamp", Unmarshaler{}, `{"ts":""}`, &pb.KnownTypes{Ts: &tspb.Timestamp{}}},
```
Comment (mariaalexandropoulou) [inline jsonpb/jsonpb_test.go:771]: Can we add a test with {"ts": null}?

Comment (cvanderschuere) [reply jsonpb/jsonpb_test.go:771]: there is one right above it

**Review by mariaalexandropoulou — APPROVED** [review]
